### PR TITLE
fix(cli/buffer): allow Buffer to store MAX_SIZE bytes

### LINKED ISSUE #6543: Buffer fails to store up to MAX_SIZE bytes
Take the following snippet, where I'm trying to read `MAX_SIZE` from a `Reader`

https://github.com/denoland/deno/blob/2da084058397efd6f517ba98c9882760ec0a7bd6/cli/js/buffer.ts#L15

```javascript
const MAX_SIZE = 2 ** 32 - 2;
const capacities = [MAX_SIZE, MAX_SIZE - 1, MAX_SIZE - 512];
for (const capacity of capacities) {
    const reader = new Deno.Buffer(new ArrayBuffer(capacity));
    const buf = new Deno.Buffer();
    buf.readFromSync(reader);
}
```

```
error: Uncaught Error: The buffer cannot be grown beyond the maximum size.
    at Buffer.#grow ($deno$/buffer.ts:137:13)
    at Buffer.readFromSync ($deno$/buffer.ts:176:27)
```

### PR DESCRIPTION
**Pull request by marcosc90**
fixes #6543

### CODE REVIEW

File: cli/js/buffer.ts [hunk 1]
```diff
@@ -157,31 +157,27 @@ export class Buffer implements Reader, ReaderSync, Writer, WriterSync {
 
   async readFrom(r: Reader): Promise<number> {
     let n = 0;
+    const buf = new Uint8Array(32 * 1024);
```
Comment (marcosc90) [inline cli/js/buffer.ts:160]: calling `this.#grow` without knowing the actual amount that needs to grow is causing troubles, went with the simpler and less error-prone code, which is calling `writeSync` which calls `this.#grow` with the size of the read buffer.

In a few minutes, I'll push an alternative reading directly to the buffer (which is not as clean). If someone else wants to try I'm more than happy.

Comment (marcosc90) [reply cli/js/buffer.ts:155]: Posted alternative implementation in #6570

**Comment by marcosc90** [comment]
In Windows, it seems the `ArrayBuffer` hard limit is different, so it can be tested correctly.

```
RangeError: Array buffer allocation failed
    at new ArrayBuffer (<anonymous>)
```

Or maybe the CI windows instance has less memory.

**Comment by marcosc90** [comment]
Ignored tests in low memory environments:

[Windows:](https://github.com/denoland/deno/pull/6568/checks?check_run_id=831865821#step:17:522)
```
test bufferGrowWriteMaxBuffer ... ignored (0ms)
test bufferGrowReadSyncCloseToMaxBuffer ... ignored (0ms)
test bufferGrowReadCloseToMaxBuffer ... ignored (0ms)
test bufferGrowReadCloseMaxBufferPlus1 ... ignored (0ms)
test bufferGrowReadSyncCloseMaxBufferPlus1 ... ignored (0ms)
test bufferReadCloseToMaxBufferWithInitialGrow ... ignored (0ms)
```

[Linux](https://github.com/denoland/deno/pull/6568/checks?check_run_id=831865828#step:17:526) and [Mac](https://github.com/denoland/deno/pull/6568/checks?check_run_id=831865807#step:17:532)
```
....
test bufferGrowReadCloseMaxBufferPlus1 ... ok (8697ms)
test bufferGrowReadSyncCloseMaxBufferPlus1 ... ok (9522ms)
test bufferReadCloseToMaxBufferWithInitialGrow ... ok (16614ms)
....
```

File: cli/tests/unit/buffer_test.ts [hunk 2]
```diff
@@ -11,11 +11,19 @@ import {
   unitTest,
 } from "./test_util.ts";
 
+const MAX_SIZE = 2 ** 32 - 2;
```
Comment (ry) [inline cli/tests/unit/buffer_test.ts:14]: Is it really necessary to allocate 4G of memory? Does the crash occur a smaller allocations?

Comment (marcosc90) [reply cli/tests/unit/buffer_test.ts:14]: I can replicate the crash on `master` using `write`/`writeSync` when writing ~2gb, but not for `readFrom`, the latter always allocate a maximum buffer of `4294966784`, 510 bytes short of MAX_SIZE

```
const buf = new Deno.Buffer()
buf.writeSync(repeat("x", 2147483647));
buf.writeSync(repeat("x", 1));
// throws The buffer cannot be grown beyond the maximum size.
```

**readFrom/Sync**

```
const reader = new Deno.Buffer(new ArrayBuffer(4294966784 + 1));
const buf = new Deno.Buffer();
buf.readFromSync(reader);
// throws The buffer cannot be grown beyond the maximum size.
```

**Comment by marcosc90** [comment]
Closing in favor of #6570

**Pull request closed without merging**